pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1021,7 +1021,7 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 	affinityErr := checkNodeAffinity(og, volumeToMount)
 	if affinityErr != nil {
 		eventErr, detailedErr := volumeToMount.GenerateError("MapVolume.NodeAffinity check failed", affinityErr)
-		og.recorder.Eventf(volumeToMount.Pod, v1.EventTypeWarning, kevents.FailedMountVolume, eventErr.Error())
+		og.recorder.Eventf(volumeToMount.Pod, nil, v1.EventTypeWarning, kevents.FailedMountVolume, "Mounting Volume", eventErr.Error())
```

Comment:
Didn't look super carefully through the whole PR, but here is ohter potential place where volume sounds like a desired related object.

Please look though all places to catch all of those.